app/assets/javascripts/components/common/ScopingMethods/autocomplete_categories_input.jsx

```diff
@@ -0,0 +1,44 @@
+import { debounce } from 'lodash';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import AsyncSelect from 'react-select/async';
+import API from '../../../utils/api';
+
+const CategoryAutoCompleteInput = ({ label, actionType, initial }) => {
+  const dispatch = useDispatch();
+  const home_wiki = useSelector(state => state.course.home_wiki);
+
+  const search = async (query) => {
+    const data = await API.getCategoriesWithPrefix(home_wiki, query);
+    return data;
+  };
+
+  const _loadOptions = (query, callback) => {
+    search(query).then(resp => callback(resp));
+  };
+  const loadOptions = debounce(_loadOptions, 300);
+
+  const updateCategories = (categories) => {
+    dispatch({
+      type: actionType,
+      categories,
+    });
+  };
+
+  return (
+    <>
+      <label htmlFor="categories">{label} </label>
+      <AsyncSelect
+        loadOptions={loadOptions}
+        placeholder={I18n.t('courses_generic.creator.scoping_methods.start_typing_to_search')}
+        isMulti
+        id="categories"
+        onChange={updateCategories}
+        noOptionsMessage={() => 'No categories found'}
```

Comment:
I notice that it's a little wonky as I start typing something, where I may see results from the first few characters, then 'No categories found' as I type more, then showing results again as I get close to the complete input. For example (on English Wikipedia):

femi -> one result
femin -> no results
feminism -> several results

It looks like it basically only matches complete words. That's fine behavior, but I wonder the 'No categories found' message might be slightly confusing.

As I've played around with it and thought through this comment, I can't think of a better way to do it. I don't think removing would be an improvement. This is fine as-is, unless you can think of a way to reduce the potential for confusion.

Reply:
have pushed a fix for this